lib/benchee/formatters/console/memory.ex

```diff
@@ -0,0 +1,199 @@
+defmodule Benchee.Formatters.Console.Memory do
+  @moduledoc """
+  This deals with just the formatting of the run time results. They are similar
+  to the way the memory results are formatted, but different enough to where the
+  abstractions start to break down pretty significantly, so I wanted to extract
+  these two things into separate modules to avoid confusion.
+  """
+
+  alias Benchee.{
+    Statistics,
+    Benchmark.Scenario,
+    Conversion,
+    Conversion.Unit,
+    Formatters.Console.Helpers
+  }
+
+  @type unit_per_statistic :: %{atom => Unit.t()}
+
+  # Length of column header
+  @average_width 15
+  @deviation_width 11
+  @median_width 15
+  @percentile_width 15
+
+  @doc """
+  Formats the memory statistics to a report suitable for output on the CLI. If
+  all memory measurements are the same and we have a standard deviation of 0.0
+  for each scenario, we don't show the statistics and report just on the single
+  measured memory usage.
+  """
+  @spec format_scenarios([Scenario.t()], map) :: [String.t(), ...]
+  def format_scenarios(scenarios, config) do
+    %{unit_scaling: scaling_strategy} = config
+    units = Conversion.units(scenarios, scaling_strategy, :memory)
+    label_width = Helpers.label_width(scenarios)
+    hide_statistics = all_have_deviation_of_0?(scenarios)
+
+    [
+      "\nMemory usage statistics:\n",
+      column_descriptors(label_width, hide_statistics)
+      | scenario_reports(scenarios, units, label_width, hide_statistics) ++
+          comparison_report(scenarios, units, label_width, config, hide_statistics)
+    ]
+  end
+
+  defp all_have_deviation_of_0?(scenarios) do
+    Enum.all?(scenarios, fn scenario ->
+      scenario.memory_usage_statistics.std_dev == 0.0
+    end)
+  end
+
+  defp column_descriptors(label_width, false) do
+    "\n~*s~*s~*s~*s~*s\n"
+    |> :io_lib.format([
+      -label_width,
+      "Name",
+      @average_width,
+      "average",
+      @deviation_width,
+      "deviation",
+      @median_width,
+      "median",
+      @percentile_width,
+      "99th %"
+    ])
+    |> to_string
+  end
+
+  defp column_descriptors(label_width, true) do
+    "\n~*s~*s\n"
+    |> :io_lib.format([
+      -label_width,
+      "Name",
+      @average_width,
+      "Memory usage"
+    ])
+    |> to_string
+  end
+
+  @spec scenario_reports([Scenario.t()], unit_per_statistic, integer, boolean) :: [String.t()]
+  defp scenario_reports([scenario | other_scenarios], units, label_width, true) do
+    [
+      reference_report(scenario, units, label_width),
+      comparisons(scenario, units, label_width, other_scenarios),
+      "\n**All measurements for memory usage were the same**\n"
```

Comment:
The only issue with that is that we have the configuration option to turn off comparisons, so if we skip the scenario reports and someone has `comparison: false` in their configuration options, nothing would be shown. We don't have any way to turn off the scenario reports, so that gets shown no matter what.

That said, your proposal does make the code more understandable. I just wonder how we'd handle that configuration clash...